### PR TITLE
ADD S13 rule and object passing and strings return design patterns

### PR DESCRIPTION
**Pull request by fgalan**
@crbrox 
@kzangeli 

Please implement directly in the PR any English language fix.

### CODE REVIEW

File: doc/manuals/contribution_guidelines.md [hunk 1]
```diff
@@ -672,3 +684,18 @@ If you plan to use some of them, please consult before with the core developers.
...
   traditional if-checking method should be avoided as much as possible.
+* Object passing. In order to avoid inefficient object copies when calling functions, the following criteria apply:
+  * If the function doesn't modify the object (i.e. "read only"), then use const references, e.g. `const BigFish& bf`
+  * If the function modifies the object (i.e. "read/write" or "write"), then use a pointer type, e.g: `BigFish* bf`. Note that in this case we could also use a reference, but we prefer pointers to make more explicit that the object is to be modified by the function.
+
+```
+void myFunction(const& BigFish in, BigFish* out)
+```
+
+* Strings return. In order to avoid inefficient object copies when returning strings, the  `const std::string&` return type is preferred. In the case that pattern cannot be used (e.g. when literal string such as "black", "red", etc. are used in the function) then `const char*` should be used as return type.
```
Comment (fgalan) [inline doc/manuals/contribution_guidelines.md]: If you find a better way of explaining in which cases `const std::string&` could be a problem and `const char*` has to be used, please go ahead.

Comment (kzangeli) [inline doc/manuals/contribution_guidelines.md]: No, this is the only case I know of thus far. Probably it is the only one where const std::string& cannot be used, as "xxx" is in a readonly segment.

Comment (kzangeli) [inline doc/manuals/contribution_guidelines.md]: NTC

**Comment by kzangeli** [comment]
LGTM

**Comment by crbrox** [comment]
LGTM

**Pull request merged**